server/listeners/eventHandlers/prisonEvents/datesChangedEventHandler.ts

```diff
@@ -47,30 +45,43 @@ export default class DatesChangedEventHandler {
 
       await Promise.all(
         licences.map(licence => {
-          return this.updateLicenceSentenceDates(licence, nomisId)
+          return this.updateLicenceSentenceDates(licence, prisoner)
         })
       )
     }
   }
 
-  deactivateLicencesIfPrisonerResentenced = async (licences: LicenceSummary[], bookingId: number) => {
+  deactivateLicencesIfPrisonerResentenced = async (licence: LicenceSummary, bookingId: number) => {
     const ssd = await this.prisonerService.getPrisonerLatestSentenceStartDate(bookingId)
-    await Promise.all(
-      licences.map(async licence => {
-        const crd = licence.conditionalReleaseDate ? parseCvlDate(licence.conditionalReleaseDate) : null
 
-        if (ssd && crd && isAfter(ssd, crd)) {
-          logger.info(
-            `new sentence start date: ${ssd} is after licence crd: ${crd} so deactivating current licence with id: ${licence.licenceId}`
-          )
-          await this.licenceService.updateStatus(licence.licenceId, LicenceStatus.INACTIVE)
-        }
-      })
-    )
+    const crd = licence.conditionalReleaseDate ? parseCvlDate(licence.conditionalReleaseDate) : null
+
+    if (ssd && crd && isAfter(ssd, crd)) {
+      logger.info(
+        `new sentence start date: ${ssd} is after licence crd: ${crd} so deactivating current licence with id: ${licence.licenceId}`
+      )
+      await this.licenceService.deactivateActiveAndVariationLicences(licence.licenceId, 'RESENTENCED')
+    }
   }
 
-  updateLicenceSentenceDates = async (licence: LicenceSummary, nomisId: string) => {
-    const prisoner = await this.prisonerService.getPrisonerDetail(nomisId)
+  deactivateLicencesIfFuturePrrd = async (licence: LicenceSummary, prisoner: PrisonApiPrisoner) => {
+    const prrd =
+      prisoner.sentenceDetail?.postRecallReleaseOverrideDate || prisoner.sentenceDetail?.postRecallReleaseDate
+    if (prrd) {
+      const prrdDate = parseIsoDate(prrd)
+      if (format(prrdDate, 'dd/MM/yyyy') === licence.postRecallReleaseDate) {
+        return
+      }
+      if (isAfter(prrdDate, startOfDay(new Date()))) {
+        logger.info(
+          `licence id: ${licence.licenceId} - deactivated due to new PRRD: ${prrd} (existing PRRD: ${licence.postRecallReleaseDate})`
```

Comment:
meh.. rather than `new PRRD`, `future PRRD`?

Reply:
I was trying to indicate that it was the updated PRRD, rather than it being in the future
I suppose that's a given though, so happy to change it

Reply:
Actually, thinking about it again, I don't think we even need this log anymore.
I added it originally as we were using a generic deactivation audit message in the API, but we now have a [recall-specific one](https://github.com/ministryofjustice/create-and-vary-a-licence-api/pull/538/files#diff-87b194beb441c7e3af0c2108a233860192fde82cbad5cc6dbebfc1069f495022R1068)